eZ/Publish/API/Repository/Tests/Values/User/Limitation/LanguageLimitationTest.php

```diff
@@ -490,6 +493,242 @@ public function testPublishVersionTranslationWhenUserHasNoAccessToAllLanguages()
+        array $limitationValues,
+        bool $containsAllTranslations
+    ): void {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+
+        $content = $this->testPrepareDataForTestsWithLanguageLimitationAndDifferentContentTranslations(
+            $limitationValues,
+            $containsAllTranslations
+        );
+
+        $location = $locationService->loadLocation($content->contentInfo->mainLocationId);
+
+        if ($containsAllTranslations) {
+            $locationService->hideLocation($location);
+            $hiddenLocation = $locationService->loadLocation($location->id);
+
+            self::assertSame(
+                $content->getVersionInfo()->languageCodes,
+                $hiddenLocation->getContent()->getVersionInfo()->languageCodes
+            );
+
+            $locationService->unhideLocation($location);
+            $revealedLocation = $locationService->loadLocation($location->id);
+
+            self::assertSame(
+                $content->getVersionInfo()->languageCodes,
+                $revealedLocation->getContent()->getVersionInfo()->languageCodes
+            );
+        } else {
+            self::expectException(UnauthorizedException::class);
+
+            $locationService->hideLocation($location);
+
+            self::expectException(UnauthorizedException::class);
+
+            $locationService->unhideLocation($location);
+        }
+    }
+
+    /**
+     * @dataProvider providerForPrepareDataForTestsWithLanguageLimitationAndDifferentContentTranslations
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     */
+    public function testPrepareDataForTestsWithLanguageLimitationAndDifferentContentTranslations(
+        array $limitationValues,
+        bool $containsAllTranslations
+    ): Content {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $permissionResolver = $repository->getPermissionResolver();
+
+        $draft = $this->createMultilingualFolderDraft($contentService);
+        $content = $contentService->publishVersion($draft->getVersionInfo());
+
+        $containsAllTranslations
+            ? self::assertEmpty(array_diff($content->getVersionInfo()->languageCodes, $limitationValues))
+            : self::assertNotEmpty(array_diff($content->getVersionInfo()->languageCodes, $limitationValues));
+
+        $permissionResolver->setCurrentUserReference(
+            $this->createEditorUserWithLanguageLimitation($limitationValues)
+        );
+
+        return $content;
+    }
+
+    public function providerForPrepareDataForTestsWithLanguageLimitationAndDifferentContentTranslations(): array
```

Comment:
Describe data sets shape, so they can be statically checked once we get phpstan on sufficient level:

```suggestion
    /**
     * @return iterable<array{array<string>, bool}>
     */
    public function providerForPrepareDataForTestsWithLanguageLimitationAndDifferentContentTranslations(): iterable
```

This describes the returned type as:
1. An `iterable` (which we can use, `array` is an `iterable`)...
2. ...which will contain two-element arrays as elements...
3. ...which will consist of an array of strings as the first element, and boolean as second.

Optionally you can add descriptive keys to data sets.